donkeycar/parts/hall_sensor_rpm.py

```diff
@@ -0,0 +1,58 @@
+"""
+Hall sensor RPM
+"""
+
+#!/usr/bin/python3
+from time import sleep
+import time, math
+
+class HallSensorRpm:
+    def __init__(self, sensor, wheel_radius):
+        import RPi.GPIO as GPIO
+        GPIO.setmode(GPIO.BCM)
+        GPIO.setwarnings(False)
+        GPIO.setup(sensor,GPIO.IN,GPIO.PUD_UP)
+        GPIO.add_event_detect(sensor, GPIO.FALLING, callback = self.calculate_elapse, bouncetime = 20)
+
+        self.rpm = 0
```

Comment:
I would probably use Hertz as a unit rather than rpm, but doesn't really matter for ML purposes.